pipeline/main.py

```diff
@@ -1,279 +0,0 @@
-                                layer_dropout=args.layer_dropout,
-                                activation=torch.tanh,
-                                batchnorm=True)
-    
-    mixhopgcn_model = mixhopgcn_model.to(device)
-
-    # Step 3: Create training components ===================================================== #
-    loss_fn = nn.CrossEntropyLoss()
-    opt = optim.SGD(mixhopgcn_model.parameters(), lr=args.lr, weight_decay=args.lamb)
-    scheduler = optim.lr_scheduler.StepLR(opt, args.step_size, gamma=args.gamma)
-
-    # Step 4: training epoches =============================================================== #
-    acc = 0
-    no_improvement = 0
-    epochs = trange(args.epochs, desc='Accuracy & Loss')
-    
-    for _ in epochs:
-
-        # Training and validation using a full graph
-        mixhopgcn_model.train()
-
-        logits = mixhopgcn_model.forward(graph, feats)
-
-        # compute loss
-        train_loss = loss_fn(logits[train_idx], labels[train_idx])
-        train_acc = torch.sum(logits[train_idx].argmax(dim=1) == labels[train_idx]).item() / len(train_idx)
-
-        valid_loss = loss_fn(logits[val_idx], labels[val_idx])
-        valid_acc = torch.sum(logits[val_idx].argmax(dim=1) == labels[val_idx]).item() / len(val_idx)
-
-        # backward
-        opt.zero_grad()
-        train_loss.backward()
-        opt.step()
-
-        # Print out performance
-        epochs.set_description('Train Acc {:.4f} | Train Loss {:.4f} | Val Acc {:.4f} | Val loss {:.4f}'.format(
-            train_acc, train_loss.item(), valid_acc, valid_loss.item()))
-        
-        if valid_acc < acc:
-            no_improvement += 1
-            if no_improvement == args.early_stopping:
-                print('Early stop.')
-                break
-        else:
-            no_improvement = 0
-            acc = valid_acc
-        
-        scheduler.step()
-
-    # Test with mini batch after all epoch
-    mixhopgcn_model.eval()
-
-    # forward
-    logits = mixhopgcn_model.forward(graph, feats)
-
-    # compute loss
-    test_loss = loss_fn(logits[test_idx], labels[test_idx])
-    test_acc = torch.sum(logits[test_idx].argmax(dim=1) == labels[test_idx]).item() / len(test_idx)
-
-    print("Test Acc {:.4f} | Test loss {:.4f}".format(test_acc, test_loss.item()))
-
-if __name__ == "__main__":
-    """
-    MixHop Model Parameters
-    """
-    parser = argparse.ArgumentParser(description='MixHop GCN')
-
-    # data source params
-    parser.add_argument('--dataset', type=str, default='Cora', help='Name of dataset.')
-    parser.add_argument('--raw-dir', type=str, default='./data', help='Path of dataset.')
-    # cuda params
-    parser.add_argument('--gpu', type=int, default=-1, help='GPU index. Default: -1, using CPU.')
-    # training params
-    parser.add_argument('--epochs', type=int, default=2000, help='Traning epochs.')
-    parser.add_argument('--early-stopping', type=int, default=200, help='Early stopping.')
```

Comment:
"Early stopping." -> "Patient epochs to wait before early stopping."

Reply:
Done.